cmd/deploy.go

```diff
@@ -101,10 +101,15 @@ Run this command from the root directory of your Appsody project.`,
 				manifestNamespace := appsodyApplication.Namespace
 				if manifestNamespace != "" {
 					if namespace != "" && manifestNamespace != namespace {
-						return errors.Errorf("the namespace \"%s\" from the deployment manifest does not match the namespace \"%s\" passed as an argument.", manifestNamespace, namespace)
+						config.Info.Logf("Overriding namespace %s in the deployment manifest to: %s", manifestNamespace, namespace)
+						appsodyApplication.Namespace = namespace
+						err = writeAppsodyApplication(appsodyApplication, configFile)
+						if err != nil {
+							return err
+						}
```

Comment:
We could move all the logic from build into here in order to have everything in one place. I think originally we put it in build so we only write once to the deployment manifest. I think putting it in here is better because in the case where users in the past passed the `--no-build` flag, the namespace won't have been updated.